Issue integrating spring-cloud-gcp-starter-sql-mysql with spring data rest

While adding spring-boot-starter-data-rest dependency to POM that has spring-cloud-gcp-starter-sql-mysql, the application is not able to start throwing the below exception. Spanner dependency is not included as the application uses vanilla JPA + GCP SQL. 

Caused by: java.lang.ClassNotFoundException: org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerMappingContext
	at java.net.URLClassLoader$1.run(URLClassLoader.java:372) ~[na:1.8.0_25]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361) ~[na:1.8.0_25]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_25]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:360) ~[na:1.8.0_25]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_25]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[na:1.8.0_25]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_25]
	... 68 common frames omitted

POMXML

<parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.0.1.RELEASE</version>
        <relativePath/> <!-- lookup parent from repository -->
    </parent>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
        <java.version>1.8</java.version>
        <spring-cloud-gcp.version>1.0.0.BUILD-SNAPSHOT</spring-cloud-gcp.version>
        <spring-cloud.version>Finchley.M9</spring-cloud.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-rest</artifactId>
        </dependency>
       <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-jpa</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.data</groupId>
            <artifactId>spring-data-rest-hal-browser</artifactId>
        </dependency>

        <!-- GCP Dependencies -->
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-gcp-starter-sql-mysql</artifactId>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
            </plugin>
        </plugins>
    </build>
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.springframework.cloud</groupId>
                <artifactId>spring-cloud-dependencies</artifactId>
                <version>${spring-cloud.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
            <dependency>
                <groupId>org.springframework.cloud</groupId>
                <artifactId>spring-cloud-gcp-dependencies</artifactId>
                <version>${spring-cloud-gcp.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>

    </dependencyManagement>

--- a/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/SpannerKeyRestSupportAutoConfiguration.java
+++ b/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/SpannerKeyRestSupportAutoConfiguration.java
@@ -19,6 +19,7 @@ package org.springframework.cloud.gcp.autoconfigure.spanner;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerMappingContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.data.rest.webmvc.spi.BackendIdConverter;
@@ -29,7 +30,7 @@ import org.springframework.data.rest.webmvc.spi.BackendIdConverter;
 
 @Configuration
 @AutoConfigureAfter(GcpSpannerAutoConfiguration.class)
-@ConditionalOnClass({ BackendIdConverter.class })
+@ConditionalOnClass({ BackendIdConverter.class, SpannerMappingContext.class})
 public class SpannerKeyRestSupportAutoConfiguration {
 
 	@Bean

